src/components/MiniPaper.vue

```diff
@@ -21,44 +21,37 @@ export default {
   watch: {
     isOpen(isOpen) {
       if (isOpen) {
-        this.scaleMiniMapToFitContent();
+        this.miniMapManager.scaleMiniMap();
       }
     },
   },
-  methods: {
-    scaleMiniMapToFitContent() {
-      setTimeout(() => {
-        this.miniPaper.scaleContentToFit({ padding: 10, maxScaleX: 0.5, maxScaleY: 0.5 });
-      });
+  computed: {
+    visibilityClass() {
+      return this.isOpen ? 'visible' : 'invisible';
+    },
+    miniMap() {
+      return this.miniMapManager.miniMapPaper;
     },
+  },
+  methods: {
     movePaper({ offsetX, offsetY }) {
-      const { x, y } = this.miniPaper.paperToLocalPoint(offsetX, offsetY);
-      const { sx, sy } = this.paper.scale();
+      const { sx: scaleX, sy: scaleY } = this.paper.scale();
       const { clientWidth, clientHeight } = this.paper.el;
-
-      this.paper.translate(
-        (clientWidth / 2) - (x * sx),
-        (clientHeight / 2) - (y * sy),
-      );
+      const { newX, newY } = this.miniMapManager.calculateNewPaperPosition(offsetX, offsetY, scaleX, scaleY, clientWidth, clientHeight);
+      this.paper.translate(newX, newY);
     },
   },
   async mounted() {
     await this.$nextTick();
 
-    this.miniPaper = new dia.Paper({
-      el: this.$refs.miniPaper,
-      model: this.graph,
-      width: 300,
-      height: 200,
-      interactive: false,
-    });
+    this.miniMapManager = MiniMapManager.factory(this.graph, this.$refs.miniPaper);
 
-    this.paper.on('render:done', this.scaleMiniMapToFitContent);
+    this.paper.on('render:done', this.miniMapManager.scaleMiniMap, this);
+    window.ProcessMaker.EventBus.$on('modeler-change', this.miniMapManager.scaleMiniMap());
```

Comment:
`$on` should accept a function, but you're _calling_ the function and so the function's return value will be passed here instead. This may be the cause of the `Error in event handler` console error.

Reply:
Right, of course. Sorry.